pallets/asset-registry/src/lib.rs

```diff
@@ -119,12 +124,11 @@ pub mod pallet {
 		) -> DispatchResult {
 			T::ReserveAssetModifierOrigin::ensure_origin(origin)?;
 
-			// verify asset is registered
+			// remove asset_id => asset_multi_location, while getting the value
 			let asset_multi_location =
-				AssetIdMultiLocation::<T>::get(asset_id).ok_or(Error::<T>::AssetIsNotRegistered)?;
-
-			// unregister asset
-			AssetIdMultiLocation::<T>::remove(asset_id);
+				AssetIdMultiLocation::<T>::mutate_exists(asset_id, Option::take)
```

Comment:
```suggestion
                 // mutate_exists(id, Option::take) reads the value and delete it in one operation. Even if it's not as clear as `get` + `remove`, it is an alternative pattern when removing items.
				AssetIdMultiLocation::<T>::mutate_exists(asset_id, Option::take)
```

Reply:
I think it's good enough, people can also read the doc for the method 😉️